.github/workflows/pitest-analyze-pr.yml

```diff
@@ -0,0 +1,32 @@
+# Performs mutation testing analysis on the files changed by a pull request and
+# uploads the results. The associated PR is subsequently updated by the
+# `pitest-update-pr.yml` workflow. See https://blog.pitest.org/oss-pitest-pr/
+# for details.
+name: "Mutation testing"
+on:
+  pull_request:
+permissions:
+  contents: read
+jobs:
+  analyze-pr:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.1.0
+        with:
+          fetch-depth: 2
+      - name: Set up JDK
+        uses: actions/setup-java@v3.6.0
+        with:
+          java-version: 17.0.4
+          distribution: temurin
+          cache: maven
+      - name: Run Pitest
+        run: mvn test pitest:mutationCoverage -DargLine.xmx=2048m -Dverification.skip -Dfeatures="+GIT(from[HEAD~1]), +gitci"
```

Comment:
At first, the `+GIT(from[HEAD~1])` triggered me thinking this would only apply on the latest commit of the feature branch, and that we instead wish to diff between the head of the target and the head of the feature branch. However, this actually refers to the entire diff as it is squashed into a single commit (see [here](https://blog.pitest.org/pitest-pr-setup/#:~:text=This%20runs%20pitest%2C%20producing%20additional%20output%20which%20is%20then%20used%20by%20the%20maven%20plugin.%20Although%20at%20first%20glance%20it%20looks%20like%20only%20the%20most%20recent%20change%20is%20analysed%2C%20in%20fact%20all%20changes%20in%20the%20PR%20will%20be%20considered%20as%20github%20places%20them%20in%20a%20single%20commit.)):

> This runs pitest, producing additional output which is then used by the maven plugin. Although at first glance it looks like only the most recent change is analysed, in fact all changes in the PR will be considered as github places them in a single commit.

Perhaps worth of a comment to avoid future confusion.

Reply:
The blogpost is mentioned in a comment on the second line, so I'm in doubt whether we need a comment on top of that. Both options are fine by me though.

Reply:
What I am linking here is a different blog post though.

Reply:
Added a commit. PTAL.

Reply:
LGTM! Nice.

Reply:
Very nice. Much more details but still concise. :+1: LGTM.